### ISSUE TITLE
Gitea / Gogs plugin

### ISSUE THREAD

**Issue by alexanderadam** [getsentry/sentry#15738]
This is the sequel of [the Sentry plugin issue 145](https://github.com/getsentry/sentry-plugins/issues/145)

## Summary

Add a plugin, that is similar to the [GitLab](https://docs.sentry.io/workflow/integrations/global-integrations/#gitlab)/[GitHub](https://docs.sentry.io/server/integrations/github/)/[BitBucket](https://docs.sentry.io/workflow/integrations/global-integrations/#bitbucket) integration, for [Gitea](https://gitea.io/)/[Gogs](https://gogs.io/).

## Motivation

[Gitea](https://gitea.io/) and [Gogs](https://gogs.io/) are git services like GitLab or Bitbucket. They are popular self-hosted and resource friendly alternatives and you will find various instances in the web.

It is obviously useful to integrate Sentry with Gitea/Gogs for companies, universities and other organisations that are hosting their instances.

Until this is fixed, no Gitea/Gogs user can't fulfil what Sentry is suggesting: [linking to a repo](https://docs.sentry.io/workflow/releases/?platform=browser#setting-up-releases).
![Screenshot_20191121_093829](https://user-images.githubusercontent.com/372620/69321212-f8d05880-0c42-11ea-93ca-bd683c2b8d3c.png)

### Gitea vs GoGS API

[**Gitea**](https://github.com/go-gitea/gitea/) is a fork of [**GoGS**](https://github.com/gogs/gogs) and Gitea has nowadays about 3'000 patches on top of GoGS, more maintainers, more functionality and bugs fixed that are still present in GoGS (that's the reason why I switched once). So if you are running in some API bug, you should keep that in mind.

### Testing & developing against an API

Also you can simply test the API integration on [the _official test instance_ at `https://try.gitea.io/`](https://try.gitea.io/). There you can also find [the _Swagger API_ definition](https://try.gitea.io/api/swagger). As you can see, JSON and HTTP are protocol wise the only requirements but [there are also various ready-to-use Swagger libraries for Python](https://swagger.io/tools/open-source/open-source-integrations/#python).

If you rather want to use a _local_ Gitea instance, you might want to use [the official _Docker image_](https://hub.docker.com/r/gitea/gitea/) (there's also a [`docker-compose.yml`](https://hub.docker.com/r/gitea/gitea/#quick-set-up) provided that you can use).

There's also more on [API authorisation in the documentation](https://docs.gitea.io/en-us/api-usage/).

**PS:** People reading this, might also be interested [in this issue (a Sentry webhook for Gitea)](https://github.com/go-gitea/gitea/issues/7548).

**Comment by manuzope** [comment]
Hey @alexanderadam, unfortunately, we don't have the bandwidth to write and support this integration today, nor do we have this on our roadmap.

You still have an option though. You could enable releases and suspect commits using [this](https://docs.sentry.io/workflow/releases/?platform=browser#alternatively-without-a-repository-integration) method.

**Comment by alexanderadam** [comment]
What a pity

**Comment by BYK** [comment]
@manuzope is this backlog then?

**Comment by aliakseiz** [comment]
Any update on this?

**Comment by sentaur-athena** [comment]
No, this is still in the backlog and not planned right now.

**Comment by holtwick** [comment]
That's really sad. They have a fine API https://docs.gitea.com/api/#tag/issue/operation/issueCreateIssue

Any way the community may contribute to this? Or is there some generic interface?

**Comment by sentaur-athena** [comment]
@holtwick good point. There are definitely ways for community to participate, especially if issue creation and syncs are the features you care most about. 

You can build an integration using Sentry's [Integration Platform](https://docs.sentry.io/organization/integrations/integration-platform/). You can also then request it to be published to be available to all Sentry users.

**Comment by sentaur-athena** [comment]
This is not in our near term roadmap.

**Comment by KaKi87** [comment]
Is this even in your long-term roadmap ?

**Comment by sentaur-athena** [comment]
@KaKi87 unfortunately this is not on roadmap as long as I can see. We have many in house integrations and their maintenance load doesn't let the team build any new integrations. I recommend following up on the ticket @holtwick created: https://github.com/go-gitea/gitea/issues/34371

We also always welcome community contributions 🙏

**Comment by tlf30** [comment]
I would love to see this implemented. Gitea now has a LOT of users and is very similar to github on the API

**Comment by Asynchronite** [comment]
I agree, even something like Forgejo could benefit from this as it is a fork of gitea.
On the backlog topic, it really is something that should be prioritized as we are seeing a giant influx of people leaving GitHub and switching to other providers.

**Comment by svr333** [comment]
Many people are making the switch (such as myself) and I would love for this to be available with Sentry.
Is this something you would accept a PR on if developed by someone outside the Sentry team?
